### PR TITLE
INSP: Trait implementation inspection

### LINKED ISSUE #3325: "Implement members" intention doesn't trigger inside the impl block
When I create a trait impl, I have this situation:
```rust
impl Default for A {
     [cursor here]
}
```
The plugin suggests to implement members but I have to move cursor up to `impl Default for A` to use the suggestion. It would be much more convenient to trigger it from where the cursor is. Note that the suggestion doesn't appear until I type `impl Default for A {` and press Enter, so I *always* have to go back.

### PR DESCRIPTION
**Pull request by Undin**
Extract some error annotations about trait implementation into separate inspection

### CODE REVIEW

File: src/main/kotlin/org/rust/ide/inspections/RsTraitImplementationInspection.kt [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+
+            val implInfo = TraitImplementationInfo.create(trait, impl) ?: return
+
+            if (implInfo.missingImplementations.isNotEmpty()) {
+                val missing = implInfo.missingImplementations.mapNotNull { it.name?.let { "`$it`" } }.joinToString(", ")
+                val endElement = impl.typeReference ?: impl.impl
+                RsDiagnostic.TraitItemsMissingImplError(impl.impl, endElement, missing, impl, Severity.ERROR)
+                    .addToHolder(holder)
+                endElement.nextSibling?.also {
+                    RsDiagnostic.TraitItemsMissingImplError(it, impl.lastChild, missing, impl, Severity.INFO)
```
Comment (Undin) [inline src/main/kotlin/org/rust/ide/inspections/RsTraitImplementationInspection.kt]: Interesting thing: if you change severity of this inspection via settings, this annotation won't change it

**Comment by vlad20012** [comment]
I have 2 similar fixes if place the caret after the type:
image

Also I don't sure that I want to see this fix inside all descendants if the impl. Maybe just in direct children of RsMembres?
image

**Comment by Undin** [comment]
@vlad20012 I've dropped changes related to additional annotation

**Comment by Undin** [comment]
bors retry

**Pull request merged**